rfd/0131-adminitrative-actions-mfa.md

```diff
@@ -0,0 +1,524 @@
+---
+authors: Brian Joerger (bjoerger@goteleport.com)
+state: draft
+---
+
+# RFD 131 - Administrative Actions MFA
+
+## Required Approvers
+
+- Engineering: @r0mant && @codingllama
+- Product: @klizhentas || @xinding33
+- Security: @jentfoo
+
+## What
+
+Enforce MFA verification for administrative actions. For example:
+
+- Adding, removing, or editing Nodes and other services.
+- Adding, removing, or editing users, roles, auth connectors, and other
+administrative resources.
+- Approving access requests.
+- Impersonating users.
+
+This will apply to administrative actions performed from any Teleport client,
+including `tctl`, `tsh`, the Teleport Web UI, and Teleport Connect.
+
+## Why
+
+Currently, Teleport only prompts for MFA verification in a few circumstances:
+
+1. User logs in with MFA.
+2. User updates their password or registered MFA devices.
+3. User connects to a service that requires Per-session MFA.
+
+As a result, the vast majority of Teleport actions do not require timely MFA
+verification. If an admin's login session is compromised, it can be used to
+perform attacks pertaining to privilege escalation, security downgrades, Auth
+connector changes, and other forms of cluster sabotage.
+
+Adding an MFA restriction to administrative actions limits this attack vector
+by re-verifying the user's identity promptly before performing any administrative
+action.
+
+## Details
+
+### UX
+
+The UX of this feature will be very similar to Per-session MFA for `tsh`, `tctl`,
+and the WebUI, and Teleport Connect.
+
+When a user performs an admin action, they will be prompted to tap their MFA key.
+Each admin action will require a single unique tap or OTP code.
+
+```console
+$ tctl rm roles/access
+Tap any security key
+# success
+```
+
+#### Editing resources
+
+With `tctl edit` or the WebUI, it is possible to edit existing resources in a
+text editor. In this case, MFA will be prompted before proceeding to the text
+editor to prevent data loss from a failed MFA challenge.
+
+```console
+$ tctl edit roles/access
+Tap any security key
+# edit role in Vim
+# success
+```
+
+### Server configuration
+
+The requirement for MFA on admin actions will be made configurable through
+`cluster_auth_preference.admin_action_mfa: off | on`.
```

Comment:
Is it possible to avoid adding extra knob? For example, in `v14` always make it optional without extra know for backwards-compatibility. In V15, make it a requirement.